lib/ClangImporter/ImportDecl.cpp

```diff
@@ -1535,7 +1535,7 @@ namespace {
         if (Name.str() == "id" && Impl.SwiftContext.LangOpts.EnableIdAsAny) {
           Impl.SpecialTypedefNames[Decl->getCanonicalDecl()] =
               MappedTypeNameKind::DoNothing;
-          return Impl.SwiftContext.getAnyDecl();
+          return nullptr;
```

Comment:
Hmm. This will break the id-as-Any work. I think we can make the 'id' typedef unavailable in Swift when the id-as-Any flag is set, using:

        auto attr = AvailableAttr::createUnconditional(
            C, "'id' is not available in Swift; use 'Any'", "", UnconditionalAvailabilityKind::UnavailableInSwift);
        Result->getAttrs().add(attr);

and then the typedef will basically disappear. This isn't an issue introduced by your change, though.